### PR TITLE
Update cli_parse plugins and testcase

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

Depends-On: https://github.com/ansible-collections/ansible.utils/pull/33
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/731
Depends-On: https://github.com/ansible-collections/ansible.netcommon/pull/223

*  Remove cli_parse action plugin as it will reside in `ansible.utils` collection going forward and add
   redirection and deprecation notice
*  Update test cases

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

cli_parse
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: galaxy.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 authors:
   - Ansible Network Community (ansible-network)
+dependencies:
+  "ansible.utils": "*"
```
Comment (pabelanger) [inline galaxy.yml:5]: If this is now a hard dependency, we need to update our jobs a bit.  Right now we have to define the dependency in zuul too,  so any reference for build-ansible-collections, that include ansible.netcommon, also need to add ansible.utils.

EG: https://github.com/ansible/ansible-zuul-jobs/blob/master/zuul.d/project-templates.yaml#L41

Once we get a buildset galaxy server in place, we don't need to do this.

**Comment by samccann** [comment]
Should this have a changelog fragment?

File: galaxy.yml (hunk 1 above)
Comment (ganeshrn) [reply galaxy.yml:5]: Yes, we plan to have `ansible.utils` as the base collection which the `ansible.netcommon` collection will depend upon.

**Comment by ganeshrn** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Review by samccann — APPROVED** [review]
docs portion lgtm

**Review by Qalthos — APPROVED** [review]

**Comment by ganeshrn** [comment]
The PR is for netcommon 2.0.0 release

**Comment by ganeshrn** [comment]
The PR can be merged for 2.0.0 release

**Review by cidrblock — APPROVED** [review]

**Comment by ganeshrn** [comment]
recheck

**Comment by ganeshrn** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by ganeshrn** [comment]
recheck

**Pull request merged**